journal.py

```diff
@@ -161,51 +172,77 @@ def read_entries(request):
 def detail_entry(request):
     """return a single entry"""
     cursor = request.db.cursor()
-    post_id = request.matchdict.get("id", None)
-    cursor.execute(READ_ENTRY, [post_id])
+    cursor.execute(SELECT_ID, (request.matchdict['id'], ))
     keys = ('id', 'title', 'text', 'created')
-    entries = [dict(zip(keys, row)) for row in cursor.fetchall()]
-    for entry in entries:
-        entry['text'] = markdown.markdown(entry['text'], extensions=['codehilite', 'fenced_code'])
-    return {'entries': entries}
+    row = cursor.fetchone()
+    entry = dict(zip(keys, row))
+    entry['text'] = markdown.markdown(
+        entry['text'], extensions=['codehilite', 'fenced_code'])
+    return {'entry': entry}
 
 
-def update_entry(request, id):
+def update_entry(request):
+    id = request.params.get('id')
     title = request.params.get('title')
     text = request.params.get('text')
-    created = datetime.datetime.utcnow()
-    request.db.cursor().execute(UPDATE_ENTRY, [title, text, created, id])
-
-
-@view_config(route_name='edit', renderer="templates/edit.jinja2")
-def edit(request):
-
-    # if.request.authenticateduserid ELSE Forbidden
-
-    id = request.matchdict['id']
-    cursor = request.db.cursor()
-    cursor.execute(READ_ENTRY, (id,))
-    keys = ('id', 'title', 'text', 'created')
-    entries = [dict(zip(keys, row)) for row in cursor.fetchall()]
-
-    if request.method == "POST":
+    request.db.cursor().execute(UPDATE_ENTRY, [title, text, id])
+
+
+@view_config(route_name='edit', renderer='json')
+def editview_entry(request):
+    if request.authenticated_userid:
+        if request.method == 'GET':
+            cursor = request.db.cursor()
+            cursor.execute(SELECT_ID, (request.params.get('id', None), ))
+            keys = ('id', 'title', 'text', 'created')
+            row = cursor.fetchone()
+            entry = dict(zip(keys, row))
+            entry['created'] = entry['created'].strftime('%b %d, %Y')
+
+            return entry
+
+        elif request.method == 'POST':
+            try:
+                update_entry(request)
+            except psycopg2.Error:
+                # this will catch any errors generated by the database
+                return HTTPInternalServerError()
+
+            cursor = request.db.cursor()
+            cursor.execute(READ_ENTRY, (request.params.get('id', None), ))
+            keys = ('id', 'title', 'text', 'created')
+            # import pdb; pdb.set_trace()
+            row = cursor.fetchone()
+            entry = dict(zip(keys, row))
+
+            entry['text'] = markdown.markdown(
+                entry['text'], extensions=['codehilite', 'fenced_code'])
+            entry['created'] = entry['created'].strftime('%b %d, %Y')
+            return entry
```

Comment:
Notice how a significant amount of the work done in the `if/elif` clauses here is actually repeated?  What could you do to reduce that repetition?